facturacion_mexico/facturacion_fiscal/timbrado_api.py

```
@@ -75,6 +75,32 @@ def resolve_cfdi_currency_exchange(sales_invoice):
+def is_nota_descuento(factura_fiscal) -> bool:
+	"""True si la Factura Fiscal Mexico es una nota de crédito por descuento/bonificación.
+
+	Se determina por el CFDI de Egreso (E) con TipoRelación 01. La intención de negocio
+	la fija el usuario en fm_tipo_nota_credito y el controlador la traduce a TipoRelación 01.
+	"""
+	tipo = (factura_fiscal.get("fm_tipo_comprobante") or "").strip()
+	rel = (factura_fiscal.get("fm_tipo_relacion_sat") or "").split(" - ")[0].strip()
+	return tipo.startswith("E") and rel == "01"
+
+
+def resolve_concepto_description(item, es_nota_descuento: bool) -> str:
```

Ruff (0.16.0)

[warning] Boolean-typed positional argument in function definition

(FBT001)